gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,905 @@
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_MISSING_DNS_NAME =
+      ZoneInfo.builder(ZONE_NAME1)
+          .description(ZONE_DESCRIPTION1)
+          .build();
+  public static final ZoneInfo ZONE_DNS_NO_PERIOD =
+      ZoneInfo.builder(ZONE_NAME1)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME_NO_PERIOD)
+          .build();
+  public static final DnsRecord A_RECORD_ZONE1 =
+      DnsRecord.builder("www." + ZONE1.dnsName(), DnsRecord.Type.A)
+          .records(ImmutableList.of("123.123.55.1"))
+          .ttl(25, TimeUnit.SECONDS)
+          .build();
+  public static final DnsRecord AAAA_RECORD_ZONE1 =
+      DnsRecord.builder("www." + ZONE1.dnsName(), DnsRecord.Type.AAAA)
+          .records(ImmutableList.of("ed:ed:12:aa:36:3:3:105"))
+          .ttl(25, TimeUnit.SECONDS)
+          .build();
+  public static final ChangeRequest CHANGE_ADD_ZONE1 = ChangeRequest.builder()
+      .add(A_RECORD_ZONE1)
+      .add(AAAA_RECORD_ZONE1)
+      .build();
+  public static final ChangeRequest CHANGE_DELETE_ZONE1 = ChangeRequest.builder()
+      .delete(A_RECORD_ZONE1)
+      .delete(AAAA_RECORD_ZONE1)
+      .build();
+
+  public static void clear() {
+    Page<Zone> zones = DNS.listZones();
+    Iterator<Zone> zoneIterator = zones.iterateAll();
+    while (zoneIterator.hasNext()) {
+      Zone zone = zoneIterator.next();
+      List<DnsRecord> toDelete = new LinkedList<>();
+      if (zone.name().startsWith(PREFIX)) {
+        Iterator<DnsRecord> dnsRecordIterator = zone.listDnsRecords().iterateAll();
+        while (dnsRecordIterator.hasNext()) {
+          DnsRecord record = dnsRecordIterator.next();
+          if (!ImmutableList.of(DnsRecord.Type.NS, DnsRecord.Type.SOA).contains(record.type())) {
+            toDelete.add(record);
+          }
+        }
+        if (!toDelete.isEmpty()) {
+          ChangeRequest deletion =
+              zone.applyChangeRequest(ChangeRequest.builder().deletions(toDelete).build());
+          checkChangeComplete(zone.name(), deletion.id());
+        }
+        zone.delete();
+      }
+    }
+  }
+
+  private static List<Zone> filter(Iterator<Zone> iterator) {
+    List<Zone> result = new LinkedList<>();
+    while (iterator.hasNext()) {
+      Zone zone = iterator.next();
+      if (zone.name().startsWith(PREFIX)) {
+        result.add(zone);
+      }
+    }
+    return result;
+  }
+
+  @BeforeClass
+  public static void before() {
+    clear();
+  }
+
+  @AfterClass
+  public static void after() {
+    clear();
+  }
+
+  private static void assertEqChangesIgnoreStatus(ChangeRequest expected, ChangeRequest actual) {
+    ChangeRequest unifiedEx = ChangeRequest.fromPb(expected.toPb().setStatus("pending"));
+    ChangeRequest unifiedAct = ChangeRequest.fromPb(actual.toPb().setStatus("pending"));
+    assertEquals(unifiedEx, unifiedAct);
+  }
+
+  private static void checkChangeComplete(String zoneName, String changeId) {
```

Comment:
I would prefer calling this something like `waitUntilComplete` to make clear this method can take a while.